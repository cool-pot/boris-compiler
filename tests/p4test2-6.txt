*** test level 2, exchange

***int
global fooi = 1;
global bari = 2;

print fooi;
print bari;
fooi <-> bari;
print fooi;
print bari;

***tuple
global foot = 1,2,3;
global bart = 2,3,4,5;
print foot;
print bart;
foot <-> bart;
print foot;
print bart;

***array
global fooa;
global bara;
array fooa[1..2];
array bara[6..8];
fooa[1] = 1;
fooa[2] = 2;
bara[6] = 6;
bara[7] = 7;
bara[8] = 8;
print fooa;
print bara;
fooa <-> bara;
print fooa;
print bara;

