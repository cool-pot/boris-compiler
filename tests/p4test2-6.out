; ModuleID = 'module-boris-input'
source_filename = "module-boris-input"
target triple = "x86_64-pc-linux-gnu"

@format_int = private unnamed_addr constant [4 x i8] c"%d\0A\00"
@format_int.1 = private unnamed_addr constant [4 x i8] c"%d\0A\00"
@format_int.2 = private unnamed_addr constant [4 x i8] c"%d\0A\00"
@format_int.3 = private unnamed_addr constant [4 x i8] c"%d\0A\00"
@format_int_0 = private unnamed_addr constant [4 x i8] c"%d,\00"
@format_int_1 = private unnamed_addr constant [3 x i8] c"%d\00"
@format_tuple_beg = private unnamed_addr constant [2 x i8] c"(\00"
@format_tuple_end = private unnamed_addr constant [3 x i8] c")\0A\00"
@format_int_0.4 = private unnamed_addr constant [4 x i8] c"%d,\00"
@format_int_1.5 = private unnamed_addr constant [3 x i8] c"%d\00"
@format_tuple_beg.6 = private unnamed_addr constant [2 x i8] c"(\00"
@format_tuple_end.7 = private unnamed_addr constant [3 x i8] c")\0A\00"
@format_int_0.8 = private unnamed_addr constant [4 x i8] c"%d,\00"
@format_int_1.9 = private unnamed_addr constant [3 x i8] c"%d\00"
@format_tuple_beg.10 = private unnamed_addr constant [2 x i8] c"(\00"
@format_tuple_end.11 = private unnamed_addr constant [3 x i8] c")\0A\00"
@format_int_0.12 = private unnamed_addr constant [4 x i8] c"%d,\00"
@format_int_1.13 = private unnamed_addr constant [3 x i8] c"%d\00"
@format_tuple_beg.14 = private unnamed_addr constant [2 x i8] c"(\00"
@format_tuple_end.15 = private unnamed_addr constant [3 x i8] c")\0A\00"
@format_int_0.16 = private unnamed_addr constant [4 x i8] c"%d,\00"
@format_int_1.17 = private unnamed_addr constant [3 x i8] c"%d\00"
@format_array_beg = private unnamed_addr constant [2 x i8] c"[\00"
@format_array_end = private unnamed_addr constant [3 x i8] c"]\0A\00"
@format_int_0.18 = private unnamed_addr constant [4 x i8] c"%d,\00"
@format_int_1.19 = private unnamed_addr constant [3 x i8] c"%d\00"
@format_array_beg.20 = private unnamed_addr constant [2 x i8] c"[\00"
@format_array_end.21 = private unnamed_addr constant [3 x i8] c"]\0A\00"
@format_int_0.22 = private unnamed_addr constant [4 x i8] c"%d,\00"
@format_int_1.23 = private unnamed_addr constant [3 x i8] c"%d\00"
@format_array_beg.24 = private unnamed_addr constant [2 x i8] c"[\00"
@format_array_end.25 = private unnamed_addr constant [3 x i8] c"]\0A\00"
@format_int_0.26 = private unnamed_addr constant [4 x i8] c"%d,\00"
@format_int_1.27 = private unnamed_addr constant [3 x i8] c"%d\00"
@format_array_beg.28 = private unnamed_addr constant [2 x i8] c"[\00"
@format_array_end.29 = private unnamed_addr constant [3 x i8] c"]\0A\00"

define i32 @strlen(i8* %s) {
init:
  %i = alloca i32
  store i32 0, i32* %i
  br label %check

check:                                            ; preds = %body, %init
  %0 = load i32, i32* %i
  %1 = getelementptr i8, i8* %s, i32 %0
  %2 = load i8, i8* %1
  %3 = icmp ne i8 0, %2
  br i1 %3, label %body, label %end

body:                                             ; preds = %check
  %4 = load i32, i32* %i
  %5 = add i32 %4, 1
  store i32 %5, i32* %i
  br label %check

end:                                              ; preds = %check
  %6 = load i32, i32* %i
  ret i32 %6
}

declare i32 @printf(...)

define void @main() {
MainEntry:
  %fooi = alloca i32
  store i32 1, i32* %fooi
  %bari = alloca i32
  store i32 2, i32* %bari
  %load_singleid_int_temp = load i32, i32* %fooi
  %0 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int, i32 0, i32 0), i32 %load_singleid_int_temp)
  %load_singleid_int_temp1 = load i32, i32* %bari
  %1 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int.1, i32 0, i32 0), i32 %load_singleid_int_temp1)
  %load_singleid_int_temp2 = load i32, i32* %bari
  %2 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int.2, i32 0, i32 0), i32 %load_singleid_int_temp2)
  %load_singleid_int_temp3 = load i32, i32* %fooi
  %3 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int.3, i32 0, i32 0), i32 %load_singleid_int_temp3)
  %foot = alloca i32, i32 3
  %4 = getelementptr i32, i32* %foot, i32 0
  store i32 1, i32* %4
  %5 = getelementptr i32, i32* %foot, i32 1
  store i32 2, i32* %5
  %6 = getelementptr i32, i32* %foot, i32 2
  store i32 3, i32* %6
  %bart = alloca i32, i32 4
  %7 = getelementptr i32, i32* %bart, i32 0
  store i32 2, i32* %7
  %8 = getelementptr i32, i32* %bart, i32 1
  store i32 3, i32* %8
  %9 = getelementptr i32, i32* %bart, i32 2
  store i32 4, i32* %9
  %10 = getelementptr i32, i32* %bart, i32 3
  store i32 5, i32* %10
  %11 = call i32 (...) @printf(i8* getelementptr inbounds ([2 x i8], [2 x i8]* @format_tuple_beg, i32 0, i32 0))
  %12 = getelementptr i32, i32* %foot, i32 0
  %13 = load i32, i32* %12
  %14 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0, i32 0, i32 0), i32 %13)
  %15 = getelementptr i32, i32* %foot, i32 1
  %16 = load i32, i32* %15
  %17 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0, i32 0, i32 0), i32 %16)
  %18 = getelementptr i32, i32* %foot, i32 2
  %19 = load i32, i32* %18
  %20 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_int_1, i32 0, i32 0), i32 %19)
  %21 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_tuple_end, i32 0, i32 0))
  %22 = call i32 (...) @printf(i8* getelementptr inbounds ([2 x i8], [2 x i8]* @format_tuple_beg.6, i32 0, i32 0))
  %23 = getelementptr i32, i32* %bart, i32 0
  %24 = load i32, i32* %23
  %25 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0.4, i32 0, i32 0), i32 %24)
  %26 = getelementptr i32, i32* %bart, i32 1
  %27 = load i32, i32* %26
  %28 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0.4, i32 0, i32 0), i32 %27)
  %29 = getelementptr i32, i32* %bart, i32 2
  %30 = load i32, i32* %29
  %31 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0.4, i32 0, i32 0), i32 %30)
  %32 = getelementptr i32, i32* %bart, i32 3
  %33 = load i32, i32* %32
  %34 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_int_1.5, i32 0, i32 0), i32 %33)
  %35 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_tuple_end.7, i32 0, i32 0))
  %36 = call i32 (...) @printf(i8* getelementptr inbounds ([2 x i8], [2 x i8]* @format_tuple_beg.10, i32 0, i32 0))
  %37 = getelementptr i32, i32* %bart, i32 0
  %38 = load i32, i32* %37
  %39 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0.8, i32 0, i32 0), i32 %38)
  %40 = getelementptr i32, i32* %bart, i32 1
  %41 = load i32, i32* %40
  %42 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0.8, i32 0, i32 0), i32 %41)
  %43 = getelementptr i32, i32* %bart, i32 2
  %44 = load i32, i32* %43
  %45 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0.8, i32 0, i32 0), i32 %44)
  %46 = getelementptr i32, i32* %bart, i32 3
  %47 = load i32, i32* %46
  %48 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_int_1.9, i32 0, i32 0), i32 %47)
  %49 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_tuple_end.11, i32 0, i32 0))
  %50 = call i32 (...) @printf(i8* getelementptr inbounds ([2 x i8], [2 x i8]* @format_tuple_beg.14, i32 0, i32 0))
  %51 = getelementptr i32, i32* %foot, i32 0
  %52 = load i32, i32* %51
  %53 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0.12, i32 0, i32 0), i32 %52)
  %54 = getelementptr i32, i32* %foot, i32 1
  %55 = load i32, i32* %54
  %56 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0.12, i32 0, i32 0), i32 %55)
  %57 = getelementptr i32, i32* %foot, i32 2
  %58 = load i32, i32* %57
  %59 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_int_1.13, i32 0, i32 0), i32 %58)
  %60 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_tuple_end.15, i32 0, i32 0))
  %fooa = alloca i32, i32 2
  %bara = alloca i32, i32 3
  %61 = getelementptr i32, i32* %fooa, i32 0
  store i32 1, i32* %61
  %62 = getelementptr i32, i32* %fooa, i32 1
  store i32 2, i32* %62
  %63 = getelementptr i32, i32* %bara, i32 0
  store i32 6, i32* %63
  %64 = getelementptr i32, i32* %bara, i32 1
  store i32 7, i32* %64
  %65 = getelementptr i32, i32* %bara, i32 2
  store i32 8, i32* %65
  %66 = call i32 (...) @printf(i8* getelementptr inbounds ([2 x i8], [2 x i8]* @format_array_beg, i32 0, i32 0))
  %67 = getelementptr i32, i32* %fooa, i32 0
  %68 = load i32, i32* %67
  %69 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0.16, i32 0, i32 0), i32 %68)
  %70 = getelementptr i32, i32* %fooa, i32 1
  %71 = load i32, i32* %70
  %72 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_int_1.17, i32 0, i32 0), i32 %71)
  %73 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_array_end, i32 0, i32 0))
  %74 = call i32 (...) @printf(i8* getelementptr inbounds ([2 x i8], [2 x i8]* @format_array_beg.20, i32 0, i32 0))
  %75 = getelementptr i32, i32* %bara, i32 0
  %76 = load i32, i32* %75
  %77 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0.18, i32 0, i32 0), i32 %76)
  %78 = getelementptr i32, i32* %bara, i32 1
  %79 = load i32, i32* %78
  %80 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0.18, i32 0, i32 0), i32 %79)
  %81 = getelementptr i32, i32* %bara, i32 2
  %82 = load i32, i32* %81
  %83 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_int_1.19, i32 0, i32 0), i32 %82)
  %84 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_array_end.21, i32 0, i32 0))
  %85 = call i32 (...) @printf(i8* getelementptr inbounds ([2 x i8], [2 x i8]* @format_array_beg.24, i32 0, i32 0))
  %86 = getelementptr i32, i32* %bara, i32 0
  %87 = load i32, i32* %86
  %88 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0.22, i32 0, i32 0), i32 %87)
  %89 = getelementptr i32, i32* %bara, i32 1
  %90 = load i32, i32* %89
  %91 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0.22, i32 0, i32 0), i32 %90)
  %92 = getelementptr i32, i32* %bara, i32 2
  %93 = load i32, i32* %92
  %94 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_int_1.23, i32 0, i32 0), i32 %93)
  %95 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_array_end.25, i32 0, i32 0))
  %96 = call i32 (...) @printf(i8* getelementptr inbounds ([2 x i8], [2 x i8]* @format_array_beg.28, i32 0, i32 0))
  %97 = getelementptr i32, i32* %fooa, i32 0
  %98 = load i32, i32* %97
  %99 = call i32 (...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @format_int_0.26, i32 0, i32 0), i32 %98)
  %100 = getelementptr i32, i32* %fooa, i32 1
  %101 = load i32, i32* %100
  %102 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_int_1.27, i32 0, i32 0), i32 %101)
  %103 = call i32 (...) @printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @format_array_end.29, i32 0, i32 0))
  br label %EndEntry

EndEntry:                                         ; preds = %MainEntry
  ret void
}
